### DISCUSSION TITLE
Difference between dump() and get() functions

### DISCUSSION THREAD

**Discussion by AndresDuque** [nlohmann/json#3776] [Q&A] [answered]
Hi there,

As the title says; what's the difference between the dump() function and get<std::string>() function? They both return a string.

Example:

object["url"].dump();
object["url"].get<std::string>()

Cheers.

**Comment by falbrechtskirchinger** [comment, marked as answer]
`dump()` serializes the entire (sub-)tree. `get<T>()` returns the node value as the given type `T` if the node value is convertible to `T`.

https://json.nlohmann.me/api/basic_json/dump/
https://json.nlohmann.me/api/basic_json/get/

**Reply by AndresDuque** [reply]
Thank you! I just solved one stuff changing from dump to get and I didn't know why as they both return one string.
